From 8c8624303873fe5d0e7e8f103730b21648a3907e Mon Sep 17 00:00:00 2001
From: Idan Kahlon <idan.kahlon@tandemg.com>
Date: Sun, 19 Sep 2021 17:19:33 +0300
Subject: [PATCH] [SW-0] bitops: fix: field_prep implementation

Since __ffs32 on preboot return the bit position as
is (without +1 like in LKD code), FIELD_PREP macro
also need to be modified accrodingly.

Change-Id: I6f28628c32c4a84c9d59f197bd5b969e6fc0f03b
Signed-off-by: Idan Kahlon <idan.kahlon@tandemg.com>
---

diff --git a/include/bitops.h b/include/bitops.h
index 901fe14..10df664 100644
--- a/include/bitops.h
+++ b/include/bitops.h
@@ -110,8 +110,6 @@
 	return num;
 }
 
-#define __bf_shf(x) (__ffs32(x) - 1)
-
 /**
  * FIELD_PREP() - prepare a bitfield element
  * @_mask: shifted mask defining the field's length and position
@@ -123,7 +121,7 @@
 #define FIELD_PREP(_mask, _val)						  \
 	(								  \
 	    {								  \
-		    ((typeof(_mask))(_val) << __bf_shf(_mask)) & (_mask); \
+		    ((typeof(_mask))(_val) << __ffs32(_mask)) & (_mask);  \
 	    }								  \
 	)
 
